pkg/transforms/policy.go

```diff
@@ -146,51 +148,79 @@ func recordRelatedObjects(c *unstructured.Unstructured, node Node) Node {
 	}
 
 	objList := make([]relatedObject, 0, len(relatedObjects))
+	nonCompliantList := []relatedObject{}
 
 	for _, item := range relatedObjects {
-		relObj := parseConfigPolicyRelatedObject(item)
+		relObj, compliant := parseConfigPolicyRelatedObject(item)
 		if relObj == nil {
 			continue
 		}
 
 		objList = append(objList, *relObj)
-	}
 
-	slices.SortFunc(objList, func(a, b relatedObject) int {
-		return strings.Compare(a.kind+a.namespace+a.name, b.kind+b.namespace+b.name)
-	})
+		if !compliant {
+			nonCompliantList = append(nonCompliantList, *relObj)
+		}
+	}
 
 	node.Metadata["relObjs"] = objList
 
+	nonComp, err := json.Marshal(nonCompliantList)
+	if err != nil {
+		glog.Error("Policy transform failed to marshal related objects", err)
+	}
+
+	node.Properties["_nonCompliantObjects"] = string(nonComp)
```

Comment:
Should this be only nonCompliant objects that are missing?

Wondering if you should filter by the reason.
```
"reason": "Resource not found but should exist",
```

Reply:
The alternative is to build the list of missing nonCompliant objects while building the edges.

[Here's where we do it for Argo Apps.](https://github.com/stolostron/search-collector/blob/main/pkg/transforms/argoapplication.go#L206-L219)

Reply:
I want the table I'm building to show whether the resource is compliant or noncompliant with the policy. So right now I'm not just using this list for the missing objects (but that is one important part of it).

I don't think I can put whether the resource is compliant or not on the resource's own properties in search, because it could be compliant with one policy, and noncompliant with another.

Reply:
Here's what I'm thinking.
- if compliant, you should have an edge linking policy and the actual resource.
- if nonCompliant and resource exists, an edge links the policy and actual resource.
- if nonCompliant and resource doesn't exist, add the missing resource in the `_nonCompliant` property.

Ping me and let's talk if it doesn't make sense.